mesonpy/__init__.py

```diff
@@ -528,7 +528,13 @@ def _bool(value: Any, name: str) -> bool:
             raise ConfigError(f'Configuration entry "{name}" must be a boolean')
         return value
 
+    def _str(value: Any, name: str) -> str:
+        if not isinstance(value, str):
+            raise ConfigError(f'Configuration entry "{name}" must be a string')
+        return value
+
     scheme = _table({
+        'cli': _str,
```

Comment:
When implementing this, I realized that it may not have to be a path to a file name - it could be an executable name as well. So both my initial check and this one are too strict. The logic should be:
- if it's a file, convert it to an absolute path
- else, assume it's an executable

Reply:
Right. If the normalization to the absolute path should be done only for the value in `pyproject.toml` you can still implement it here:
```python
def _string_or_path(value: Any, name: str) -> str:
    if not isinstance(value, str):
        raise ConfigError(f'Configuration entry "{name}" must be a string')
    path = patlib.Path(value)
    if path.is_file():
        path = path.resolve()
    return str(path)
```
or maybe without round tripping through `pathlib.Path`:

```python
def _string_or_path(value: Any, name: str) -> str:
    if not isinstance(value, str):
        raise ConfigError(f'Configuration entry "{name}" must be a string')
    if os.path.isfile(value):
        value = os.path.abspath(value)
    return value
```